### PR TITLE
chore(test): Add npm test tasks into extensions to be run individually

### LINKED ISSUE #1766: Include test tasks and configure coverage options for running npm target from inside extensions module
### Is your enhancement related to a problem? Please describe

Missing test command inside particular extensions and option to also see a coverage when developing particular extension.

### Describe the solution you'd like

Include yarn/npm task `test` and configure coverage options when insider of extension. Ie. `npm run test` when inside `extensions/kind`. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Allows to run `yarn test` inside every extension module
### Screenshot/screencast of this PR
```
$ yarn test
yarn run v1.22.19
warning compose@0.0.1: The engine "podman-desktop" appears to be invalid.
$ npm run build && vitest run -r src --coverage

> compose@0.0.1 build
> vite build && node ./scripts/build.js

vite v4.2.1 building for production...
✓ 152 modules transformed.
dist/extension.js  1,837.04 kB │ map: 754.55 kB
✓ built in 1.44s

 RUN  v0.29.7 /home/odockal/git/podman-desktop/extensions/compose/src
      Coverage enabled with c8

stdout | compose-extension.spec.ts > deactivate
stopping compose extension

 ✓ detect.spec.ts (10) 513ms
 ✓ compose-wrapper-generator.spec.ts (4)
 ✓ os.spec.ts (3)
 ✓ compose-github-releases.spec.ts (10)
 ✓ compose-extension.spec.ts (16)

 Test Files  5 passed (5)
      Tests  43 passed (43)
   Start at  07:43:24
   Duration  1.24s (transform 487ms, setup 0ms, collect 820ms, tests 628ms)

 % Coverage report from c8
------------------------------|---------|----------|---------|---------|-------------------
File                          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------------------|---------|----------|---------|---------|-------------------
All files                     |    75.1 |    97.22 |   93.75 |    75.1 |                   
 cli-run.ts                   |       0 |        0 |       0 |       0 | 1-119             
 compose-extension.ts         |     100 |      100 |     100 |     100 |                   
 compose-github-releases.ts   |     100 |      100 |     100 |     100 |                   
 compose-wrapper-generator.ts |     100 |      100 |     100 |     100 |                   
 detect.ts                    |     100 |      100 |     100 |     100 |                   
 extension.ts                 |       0 |        0 |       0 |       0 | 1-54              
 os.ts                        |     100 |      100 |     100 |     100 |                   
------------------------------|---------|----------|---------|---------|-------------------
Done in 4.16s.
```

### What issues does this PR fix or reference?
Closes #1766 

### How to test this PR?
```
cd extensions/compose
yarn test
```

### CODE REVIEW

File: extensions/compose/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
   },
   "scripts": {
     "build": "vite build && node ./scripts/build.js",
+    "test": "npm run build && vitest run -r src --coverage",
```
Comment (benoitf) [inline extensions/compose/package.json]: I would only do the vitest part, not the build

People can do `npm build` &&  `npm test`

Comment (benoitf) [inline extensions/compose/package.json]: probably need a `test:watch` as well for development mode

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
it looks like you have a conflicting file

**Comment by odockal** [comment]
@benoitf Should be ok now.

**Pull request merged**